src/Utilities/promote_operation.jl

```diff
@@ -0,0 +1,426 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    promote_operation(
+        op::Function,
+        ::Type{T},
+        ArgsTypes::Type{<:Union{T,AbstractVector{T},MOI.AbstractFunction}}...,
+    ) where {T}
+
+Compute the return type of the call `operate(op, T, args...)`, where the types
+of the arguments `args` are `ArgsTypes`.
+
+There are five methods for which we implement `Utilities.promote_operation`:
+
+ 1. `+`
+    a. `promote_operation(::typeof(+), ::Type{T}, ::Type{F1}, ::Type{F2})`
+ 2. `-`
+    a. `promote_operation(::typeof(-), ::Type{T}, ::Type{F})`
+    b. `promote_operation(::typeof(-), ::Type{T}, ::Type{F1}, ::Type{F2})`
+    c. `promote_operation(::typeof(-), ::Type{T}, ::Type{F1}, ::Type{Vector{T}})`
```

Comment:
Why not 
```julia
`promote_operation(::typeof(-), ::Type{T}, ::Type{F1}, ::Type{T})`
```
then ?

Reply:
We actually did't need 2c as a case. It was already covered by 2b.